File: palindrome-partitioning/palindrome-partitioning.cs
public class Solution {
    List<IList<string>> result;
    public IList<IList<string>> Partition(string s) {
        result = new List<IList<string>>();
        if(s.Length==0 || s==null) return result;
        
        helper(s,new List<string>(),0);
        return result;
        
    }
    private void helper(string s,List<string> temp, int index)
    {
        if(index==s.Length)
        {
            result.Add(new List<string>(temp));
            return;
        }
        
        for(int i=index;i<s.Length;i++)
        {
            if(ispalindrome(s,index,i))
            {
                Console.WriteLine(index);
                Console.WriteLine(i+1);
                temp.Add(s.Substring(index,i-index+1));
                helper(s,temp,i+1);
                temp.RemoveAt(temp.Count-1);
            }
        }
    }
    
    private bool ispalindrome(string s,int low,int high)
    {
        while(low<high)
        {
            if(s[low++] != s[high--]) return false;
        }
        return true;
    }
}

File: .idea/PalindromePartitioning.java
class PalindromePartitioning {
    List<List<String>> result = new ArrayList<>();
    public List<List<String>> partition(String s) {
        if(s==null || s.length()==0)
            return result;
        backtrack(s,0, new ArrayList());
        return result;

    }
    private void backtrack(String s, int index, List<String> path)
    {
        if(s.length()==index)
        {
            result.add(new ArrayList<>(path));
            return;
        }

        //logic
        for(int i =index; i<s.length();i++)
        {
            if(isPalindrome(s, index,i))
            {
                path.add(s.substring(index,i+1));
                backtrack(s, i+1,path);
                path.remove(path.size()-1);
            }
        }

    }
    private boolean isPalindrome(String s, int l, int r)
    {
        if(l==r)
            return true;
        while(l<r)
        {
            if(s.charAt(l)!=s.charAt(r)) return false;
            l++; r--;
        }
        return true;
    }
}